### PR TITLE
chrome: improvements to the python launch script

### PR DESCRIPTION
**Pull request by pohutukawa**
chrome: improvements to the python launch script

- improved exception reporting to the logger
- improved subprocess handling to be more pythonic
- fixed compatibility with configuration for custom audio devices (ALSA) to be compatible with previous custom settings
- fixed launchability of Chrome in presence of custom audio device config (that were working previously)
- slight simplifications for more pythonic code in some functions

### CODE REVIEW

File: packages/addons/browser/chrome/source/default.py [hunk 1]
```diff
@@ -74,28 +87,30 @@ def getAudioDevice():
                    })))['result']['value']
 
 def getAudioDeviceType(dev):
-  if dev.startswith("ALSA:"):
+  if dev.startswith("ALSA:") or dev.startswith('plughw:'):
```
Comment (mglae) [inline packages/addons/browser/chrome/source/default.py:90]: That will only solve your use case. For maximum backward compatibility the algorithm should be: if dev is not empty and does not start with "ALSA:" or "PULSE:" it is ALSA.

Comment (pohutukawa) [reply packages/addons/browser/chrome/source/default.py]: @mglae I'm not really sure how to interpret the logic in your expression. Have a look whether my amended (thanks @chewitt !) PR may be more suitable.

File: packages/addons/browser/chrome/source/default.py [hunk 2]
```diff
@@ -74,28 +87,31 @@ def getAudioDevice():
...
-  if dev.startswith("ALSA:"):
-    return "ALSA"
-  if dev.startswith("PULSE:"):
-    return "PULSE"
-  return None
+  if dev:
+    if dev.startswith("ALSA:") or dev.startswith('plughw:'):
+      return "ALSA"
+    elif dev.startswith("PULSE:"):
+      return "PULSE"
```
Comment (mglae) [inline packages/addons/browser/chrome/source/default.py:94]: In the past every custom device was an alsa one (had to open a new comment for the suggestion):
```suggestion
    if dev.startswith("ALSA:"):
      return "ALSA"
    elif dev.startswith("PULSE:"):
      return "PULSE"
    else:
      # Backward compatibility of device string
      return "ALSA"
```

Comment (pohutukawa) [reply packages/addons/browser/chrome/source/default.py:94]: Got it now, thanks. Will provide the change in an update to this PR.

Comment (pohutukawa) [reply packages/addons/browser/chrome/source/default.py]: OK, updated the PR according to your suggestion @mglae

**Comment by pohutukawa** [comment]
Also created a PR for the backport to the `libreelec-10.0` branch in PR #6009

File: packages/addons/browser/chrome/source/default.py [hunk 3]
```diff
@@ -74,28 +87,34 @@ def getAudioDevice():
...
+    else:
+      # Backward compatibility of device string.
+      return "ALSA"
+  return "ALSA"
 
 def getAlsaAudioDevice(dev):
+  if dev.startswith('plughw:'):
```
Comment (mglae) [inline packages/addons/browser/chrome/source/default.py:101]: Here is a similar test required:
```suggestion
  if not dev.startswith('ALSA:'):
```

**Comment by pohutukawa** [comment]
Sorry for the delay, but I've been on a camping holiday and recently got back. It's hot and summery here ... :-)

I've incorporated the suggested change from above.

**Comment by mglae** [comment]
Thanks. Unfortunately this PR needs a rebase to current master and revision 112 already exist.

**Comment by pohutukawa** [comment]
What package version should it be, then? 112 or 113? (As the `libreelec-10.0` doesn't have a 112, yet, so my worry is that they won't be in sync).
Dang, I wish I had gotten this sorted out before the holiday :-)

**Comment by chewitt** [comment]
The `master` and `libreelec-10.0` branches are not in-sync to start with (each contains different overall content) so there is no need for the PRs to be in-sync. Just update the `master` branch PR to the next available number.

**Comment by pohutukawa** [comment]
done. I just don't know how/what those two conflicts above are coming from, as I guess they should be part of the PR on the change. Sorry for being such a nuisance ... :-(

**Comment by mglae** [comment]
You have directly edited and pushed the `master` branch. It cannot updated any more following the usual workflow. Fortunately the local and remote branch name can be different.

To clean up your local master branch environment use (untested):

```
# Create a backup. Just in case.
git branch -c master le11_chrome_backup_220127

# Create the new working branch
git branch -c master le11_chrome_improvements

# Reset and update local master branch to LE master
git checkout master
git fetch upstream
git reset --hard upstream/master

# Rebase working branch
git checkout le11_chrome_improvements
git rebase -i master

# When all done force push the working branch to the existing PR
git push -f origin le11_chrome_improvements:master
```
Similar has to be done for the backport but let's start with master.

**Comment by heitbaum** [comment]
Work incorporated into
- #7101
- #7310

**Pull request closed without merging**